pallets/pallet-bonded-coins/src/tests/transactions/start_refund.rs

```diff
@@ -0,0 +1,307 @@
+			assert_err!(
+				BondingPallet::force_start_refund(origin, pool_id, currency_count),
+				BadOrigin
+			);
+		});
+}
+
+#[test]
+fn start_refund_fails_when_no_permission() {
+	let pool_details = generate_pool_details(
+		vec![DEFAULT_BONDED_CURRENCY_ID],
+		get_linear_bonding_curve(),
+		false,
+		Some(PoolStatus::Active),
+		Some(ACCOUNT_00),
+		None,
+		None,
+	);
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+	let currency_count = 10;
+
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_00, u128::MAX)])
+		.with_pools(vec![(pool_id.clone(), pool_details.clone())])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.with_bonded_balance(vec![
+			(DEFAULT_COLLATERAL_CURRENCY_ID, pool_id.clone(), u128::MAX),
+			(DEFAULT_BONDED_CURRENCY_ID, ACCOUNT_00, u128::MAX),
+		])
+		.build()
+		.execute_with(|| {
+			let origin = RawOrigin::Signed(ACCOUNT_01).into();
+
+			// Ensure the start_refund call fails due to ALICE not having permission
+			assert_err!(
+				BondingPallet::start_refund(origin, pool_id, currency_count),
+				Error::<Test>::NoPermission
+			);
+		});
+}
+
+#[test]
+fn start_refund_fails_when_nothing_to_refund() {
+	let pool_details = generate_pool_details(
+		vec![DEFAULT_BONDED_CURRENCY_ID],
+		get_linear_bonding_curve(),
+		false,
+		Some(PoolStatus::Active),
+		Some(ACCOUNT_00),
+		None,
+		None,
+	);
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+	let currency_count = 10;
+
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_00, u128::MAX)])
+		.with_pools(vec![(pool_id.clone(), pool_details.clone())])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.with_bonded_balance(vec![(DEFAULT_COLLATERAL_CURRENCY_ID, pool_id.clone(), u128::MAX)])
+		.build()
+		.execute_with(|| {
+			let origin = RawOrigin::Signed(ACCOUNT_00).into();
+
+			// Ensure the start_refund call fails due to nothing to refund
+			assert_err!(
+				BondingPallet::start_refund(origin, pool_id, currency_count),
+				Error::<Test>::NothingToRefund
+			);
+		});
+}
+
+#[test]
+fn start_refund_fails_when_no_collateral() {
+	let pool_details = generate_pool_details(
+		vec![DEFAULT_BONDED_CURRENCY_ID],
+		get_linear_bonding_curve(),
+		false,
+		Some(PoolStatus::Active),
+		Some(ACCOUNT_00),
+		None,
+		None,
+	);
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+	let currency_count = 10;
+
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_00, u128::MAX)])
+		.with_pools(vec![(pool_id.clone(), pool_details.clone())])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.with_bonded_balance(vec![(DEFAULT_BONDED_CURRENCY_ID, ACCOUNT_00, u128::MAX)])
+		.build()
+		.execute_with(|| {
+			let origin = RawOrigin::Signed(ACCOUNT_00).into();
+
+			// Ensure the start_refund call fails due to nothing to refund
+			assert_err!(
+				BondingPallet::start_refund(origin, pool_id, currency_count),
+				Error::<Test>::NothingToRefund
+			);
+		});
+}
```

Comment:
Maybe one more test, if the pool is unknown?

Reply:
d7a398f16a6fdadb733b1d57920454fa96499fcc